client/verta/verta/_blob.py

```diff
@@ -13,7 +13,7 @@ def __eq__(self, other):
         if not isinstance(other, type(self)):
             return NotImplemented
 
-        return self._as_proto() == other._as_proto()
+        return repr(self) == repr(other)
```

Comment:
Can you share the motivated this change?

Reply:
Ah, yes!

What I specifically encountered was that when the client creates a new e.g. `Python()`, it doesn't necessarily sort the requirements before sending a creation request to the backend. But when the backend receives it, it sorts e.g. [`repeated PythonRequirementEnvironmentBlob requirements`](https://github.com/VertaAI/modeldb/blob/b497e4b8b82aba8557dd24145b3ef1986a5a9830/protos/protos/public/modeldb/versioning/Environment.proto#L41) before saving the blob to the database. Since the field is a list rather than a mapping, the difference in ordering affects the protos' equality.

To not be beholden to differences in internal representation that may be out of the client's control, I changed this to use the objects' `repr` for which the client has already bundled logic for sorting things (for user-friendly consistency).